src/pages/[slugParent]/[[...slugChild]].js

```diff
@@ -0,0 +1,181 @@
+        <Section>
+          <Container>
+            <div
+              className={styles.content}
+              dangerouslySetInnerHTML={{
+                __html: content,
+              }}
+            />
+          </Container>
+        </Section>
+
+        {hasChildren && (
+          <Section className={styles.sectionChildren}>
+            <Container>
+              <aside>
+                <p className={styles.childrenHeader}>
+                  <strong>{title}</strong>
+                </p>
+                <ul>
+                  {children.map((child) => {
+                    return (
+                      <li key={child.id}>
+                        <Link href={pagePathBySlug(`${slug}/${child.slug}`)}>
+                          <a>{child.title}</a>
+                        </Link>
+                      </li>
+                    );
+                  })}
+                </ul>
+              </aside>
+            </Container>
+          </Section>
+        )}
+      </Content>
+    </Layout>
+  );
+}
+
+export async function getStaticProps({ params = {} } = {}, ...rest) {
+  const { slugParent, slugChild } = params;
+
+  const { pages } = await getAllPages();
+
+  const id = pages.find(({ slug, parent }) => {
+    // If there's a child parameter, we need ot make sure we're matching
+    // both that we're on the right slug under the right parent
+
+    if (slugChild) {
+      // This logic only supports a 1 level deep slug, where we grab the
+      // last slug in the params and make sure it, and it's parent, matches
+      // our parameters.
+
+      const lastChild = slugChild[slugChild.length - 1];
+
+      return slug === lastChild && parent.slug === slugParent;
+    }
+
+    // If there's no child parameter, it's a top level page
+
+    return slug === slugParent;
+  })?.id;
+
+  const { page } = await getPageById(id);
+
+  return {
+    props: {
+      page,
+    },
+  };
+}
+
+export async function getStaticPaths() {
+  const routes = {};
+
+  const { pages } = await getAllPages();
+
+  // The path matching with Next.js appears to want us to separately specify a top level route
+  // along with it's child routes in order for both to be available within a catch all.
+
+  // First we build the tpo level pages where we wouldnt have a secondary param
+
+  const parentPages = pages.map((page) => {
+    const { slug, children } = page;
+    return {
+      params: {
+        slugParent: slug,
+        slugChild: [],
+      },
+    };
+  });
+
+  // Find any pages that have children
+
+  const pagesWithChildren = pages.filter(({ children }) => Array.isArray(children) && children.length > 0);
+
+  // Then map through those pages to create an additional set of paths for our pages along
+  // with their child slugs
+
+  const childPages = pagesWithChildren.map((page) => {
+    const { slug, children } = page;
+    return {
+      params: {
+        slugParent: slug,
+        slugChild: children.map(({ slug }) => slug),
```

Comment:
good catch - i started trying t ogo down the rabbit hole of getting it working but i noticed that /getting-started would work too which isn't correct

i found though in the request, the pages have a `uri` https://developer.wordpress.org/reference/functions/get_page_uri/

for pages, it seems like i might be safe using this

![image](https://user-images.githubusercontent.com/1045274/107293650-09ccdb80-6a3a-11eb-9aff-374ee529e03a.png)


i dont think i would be able to use it for posts, but since page uris seem to be the relationship of the parent / children, seems safe

going to go down that route, lmk if that doesnt seem right to you

Reply:
it would make this all a lot easier :)

Reply:
That looks like an optimal solution!